include/silo/append/database_inserter.h

```diff
@@ -0,0 +1,88 @@
+#pragma once
+
+#include <memory>
+
+#include "silo/append/append_exception.h"
+#include "silo/database.h"
+#include "silo/storage/table_partition.h"
+
+namespace silo::append {
+
+class TablePartitionInserter {
+   std::shared_ptr<storage::TablePartition> table_partition;
+
+  public:
+   class Commit {
+      friend class TablePartitionInserter;
+
+      Commit() = default;
+   };
+
+   TablePartitionInserter(std::shared_ptr<storage::TablePartition> table_partition)
+       : table_partition(table_partition) {}
+
+   void insert(const nlohmann::json& ndjson_line);
+
+   Commit commit();
+};
+
+class TableInserter {
+   storage::Table* table;
+
+  public:
+   class Commit {
+      friend class TableInserter;
+
+      Commit() = default;
+   };
+
+   TableInserter(storage::Table* table)
+       : table(table) {}
+
+   TablePartitionInserter openNewPartition();
+
+   Commit commit();
+};
+
+template <std::ranges::range Data>
+   requires std::same_as<std::ranges::range_value_t<Data>, nlohmann::json>
+silo::append::TablePartitionInserter::Commit appendDataToTablePartition(
+   silo::append::TablePartitionInserter partition_inserter,
+   Data input_data
+) {
+   size_t line_count = 0;
+   for (const auto& json_obj : input_data) {
+      partition_inserter.insert(json_obj);
```

Comment:
`json_obj` is a `nlohmann::json`, isn't it? Why don't we directly ask it to convert itself to a proper struct that reflects the structure that we expect?
```suggestion
   for (const auto& json_obj : input_data) {
      partition_inserter.insert(json_obj.get<DataEntry>());
```

where 
```c++
struct DataEntry {
   map<...> metadata;
   map<...> aligned_nucleotide_sequences;
   // ...
}
``` 

Then we could also distinguish better between error (whether the line had a wrong format or whether the data in a line was wrong)

Reply:
While true, this would mean that we copy all sequence data one additional time into that struct, and also adds more boiler-plate which makes this code more rigid. I am not a big fan of the current ndjson format, and maybe we want to change it in the future?

Reply:
Copying is a valid point, we probably want to avoid that. The data is potentially gigantic.

I think "rigidity" wouldn't hurt here. We have that rigidity already anyway since it's the same JSON structure that we're dealing with (no matter whether as a struct or as plain `nlohmann::json`). Having it in a struct makes it more transparent though.
Changing the NDJSON format later would require changes in any case, but having a struct the compiler could help.